tiledb/sm/array/array.cc

```diff
@@ -205,7 +205,7 @@ void Array::create(
 
   std::lock_guard<std::mutex> lock{object_mtx};
   array_schema->set_array_uri(array_uri);
-  array_schema->generate_uri();
+  array_schema->generate_uri(array_schema->timestamp_range());
```

Comment:
Nothing to do in this PR, but I'm not a fan of storing URIs in the array schema object. And the call to `generate_uri` seems like a code smell to me (because we call a method on another object with parameters derived from that object, and return nothing).